asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
That's interesting. I suggest handling those cases differently. Attempting to add an assertion to the database that matches the existing revision should be silently accepted if the assertion content matches exactly (it's a no-op, and the end result is the same, idempotent), but if the assertion doesn't match it should be a harsh failure rather than pretending the assertion is simply not up-to-date enough. Having two conflicting statements with the same revision seems like something is either seriously broken, or someone is trying to maliciously abuse the system. Both are cases worth raising prominently, rather than pretending it's just an old assertion we don't care about.

What do you think?